app/src/main/java/com/tolkiana/tacoapp/TacoListActivity.kt

```diff
@@ -13,8 +11,9 @@ class TacoListActivity : AppCompatActivity() {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_taco_list)
 
-        val tacoList = DataParser().parseArrayFromAsset<Product>(this.assets, "tacos.json")
-        val productAdapter = ProductAdapter(tacoList)
+        val tacoApplication = application as TacoApplication
+        val tacoList = tacoApplication.dataParser?.parseArrayFromAsset<Product>(this.assets, "tacos.json")
```

Comment:
If `DataParser` is an `object` Singleton, you'll no longer have to reference it from the application class and won't have to use the `?` keyword, since `DataParser` will never be null.